libcni/api.go

```diff
@@ -391,6 +406,65 @@ func (c *CNIConfig) GetNetworkCachedConfig(net *NetworkConfig, rt *RuntimeConf)
 	return c.getCachedConfig(net.Network.Name, rt)
 }
 
+// GetCachedAttachments returns a list of network attachments from the cache.
+// The returned list will be filtered by the containerID if the value is not empty.
+func (c *CNIConfig) GetCachedAttachments(containerID string) ([]*NetworkAttachment, error) {
```

Comment:
Does it make sense to have a collection of NetworkAttachments or some type that has a parent with multiple attachment type children? Some of the properties won't change. The netns property could be in the parent type and then the properties that are attachment specific in the children?

Reply:
Yes it does make sense in higher level packages like `go-cni` and `ocicni`, each of them have similar constructs like `Namespace` and `PodNetwork`, both concern at the level of container/pod/ns, and have container id, netns, and list of attachments as properties. `libcni` works at lower level and mostly concerns about individual attachments. IMO in the current model it's better to keep the higher level constructs in those packages.